File: CH4. Implementation/4-1.py
# 4-1-1. 상하좌우


n = int(input())

moves = input().split()


def solution(moves, n):
    x, y = 1, 1

    # L(eft), R(ight), U(p), D(own)
    directions = ["L", "R", "U", "D"]

    dx = [0, 0, -1, 1]  # x축 좌우 이동
    dy = [-1, 1, 0, 0]  # y축 상하 이동

    for move in moves:

        # 이동 방향 확인
        for i in range(0, 4):
            if move == directions[i]:

                # 이동 후, 공간 안에 있는지 확인
                nx = x + dx[i]
                ny = y + dy[i]

                if (1 <= nx <= n) and (1 <= ny <= n):
                    x = nx
                    y = ny


    print(x, y)


solution(moves, n)

File: CH 4. Implementation/4-1.js
// 4-1. 상하좌우


function solution(moves, n) {
    var x = 1;
    var y = 1;

    // L(eft), R(ight), U(p), D(own)
    const directions = ["L", "R", "U", "D"];

    const dx = [0, 0, -1, 1];
    const dy = [-1, 1, 0, 0];


    moves = moves.split(" ");


    for (var i = 0; i < moves.length; i++) {

        // 이동 방향 확인
        for (var j = 0; j < 4; j++) {
            if (moves[i] === directions[j]) {

                // 이동 후, 공간 안에 있는지 확인
                var nx = x + dx[j];
                var ny = y + dy[j];

                if (nx >= 1 && nx <= n && ny >= 1 && ny <= n) {
                    x = nx;
                    y = ny;
                }
            }
        }
    }

    console.log(x, y)
}


solution("R R R U D D", 5)